Modules/Shop/Http/Services/ShopService.php

```diff
@@ -130,54 +127,54 @@ public function getBrandProducts($request): array
 
             $products = $allProductQuery->get();
             if ($products) {
-                foreach ($products as $v) {
-                    $stock = $v->stock;
-                    $usdWholesalePrice = $v->usd_wholesale_price ?? 0;
-                    $usdRetailPrice = $v->usd_retail_price ?? 0;
-                    $productOptionsCount = ProductVariation::where('product_id', $v->id)->where('status', '1')->count();
+                foreach ($products as $productDetails) {
+                    $stock = $productDetails->stock;
+                    $usdWholesalePrice = $productDetails->usd_wholesale_price ?? 0;
+                    $usdRetailPrice = $productDetails->usd_retail_price ?? 0;
+                    $productOptionsCount = Product::find($productDetails->id)->productVariations()->count();
                     if ($productOptionsCount > 0) {
-                        $productOptionsCount = ProductVariation::where('product_id', $v->id)->where('status', '1')->sum('stock');
+                        $productOptionsCount = Product::find($productDetails->id)->productVariations()->where('status', '1')->sum('stock');
                         $stock = $productOptionsCount;
-                        $productFirstVariation = ProductVariation::where('product_id', $v->id)->where('status', '1')->first();
+                        $productFirstVariation = Product::find($productDetails->id)->productVariations()->where('status', '1')->first();
```

Comment:
All there queries where you're getting the productVariations, get the productVariations when you get the product list. and in these queries use it like $productDetails->productVariations()->further query....
I have given you these suggestions in previous PRs, please follow those in next one's.